### DISCUSSION TITLE
How do I refer to env context in a workflow file?

### DISCUSSION THREAD

**Discussion by jamiet-msm** [community/community#25759] [Actions]
<p>I’m banging my head against the desk trying to figure out the correct syntax for my workflow YAML. I am using a container on a self-hosted runner and I would like to specify the hostname for the container. This works:</p>
<pre><code class="lang-yaml">jobs:
  myjob:
    runs-on: [ self-hosted, ubuntu, dataplatform ]
    container:
      image: image:tag
      options: --hostname myhostname
</code></pre>
<p>However I want to specify it dynamically by setting hostname to the value of GITHUB_RUN_NUMBER and I can’t figure out how to do it.</p>
<p>This:</p>
<pre><code class="lang-yaml">jobs:
  myjob:
    runs-on: [ self-hosted, ubuntu, dataplatform ]
    container:
      image: image:tag
      options: --hostname $GITHUB_RUN_NUMBER
</code></pre>
<p>just sets hostname to the literal value “$GITHUB_RUN_NUMBER”, here is what gets returned from running  <code>docker inspect</code> on the container</p>
<p><img src="https://user-images.githubusercontent.com/79862277/181093136-83ba35cb-3a50-4c2c-9097-06babdaf841e.png" alt="image" data-base62-sha1="3Nv8on2SyD52aDkLlQF6UUqybFQ" width="359" height="83"></p>
<p>So I tried this instead:</p>
<pre><code class="lang-yaml">jobs:
  myjob:
    runs-on: [ self-hosted, ubuntu, dataplatform ]
    container:
      image: image:tag
      options: --hostname ${{ env.GITHUB_RUN_NUMBER }}
</code></pre>
<p>but that just caused an error when attempting to start the workflow:</p>
<blockquote>
<p>[ <strong>Invalid workflow file</strong> : .github/workflows/demo.yml#L29]<br>
The workflow is not valid. .github/workflows/demo.yml (Line: 29, Col: 16): Unrecognized named-value: ‘env’. Located at position 1 within expression: env.GITHUB_RUN_NUMBER</p>
</blockquote>
<p>Please can someone tell me the correct syntax to refer to env vars in <code>jobs.&lt;job_id&gt;.container.options</code>?</p>

**Comment by airtower-luna** [comment]
<p>You don’t. The <a href="https://docs.github.com/en/actions/learn-github-actions/contexts#env-context"><code>env</code> context</a> isn’t available outside steps:</p>
<blockquote>
<p>You can use the <code>env</code> context in the value of any key in a <strong>step</strong> except for the <code>id</code> and <code>uses</code> keys.</p>
</blockquote>

**Comment by jamiet-msm** [comment, marked as answer]
<p>got it, thank you. I have gotten around this by providing the GITHUB_RUN_NUMBER as an output from a previous job:</p>
<pre><code class="lang-yaml">  github-run-number:
    name: Provide GITHUB_RUN_NUMBER
    outputs:
      github-run-number: ${{ steps.github-run-number.outputs.github-run-number }}
    steps:
      - id: github-run-number
        run: |
          echo "::set-output name=github-run-number::$GITHUB_RUN_NUMBER"
</code></pre>
